ComponentKit/Core/ComponentUtilities.h

```diff
@@ -75,6 +75,31 @@ namespace CK {
 
     return newVector;
   }
+
+  template <typename T, typename Func, typename Predicate>
+  auto intersperse(std::vector<T> a, Func &&factory, Predicate predicate) -> std::vector<T>
```

Comment:
Also, it isn't entirely obvious what this is supposed to do. A brief comment perhaps?